transformer_engine/common/comm_gemm_overlap/userbuffers/userbuffers-host.cpp

```diff
@@ -416,6 +416,8 @@ int create_communicator_grouped2_mpi(communicator **comm, int pipegpus, int pipe
   // find internode numbers and make internode communicator
   NVTE_CHECK_CUDA(cudaFree(0));
   int mynode, numnodes;
+  mynode = myrank / numlocal;
+  numnodes = numranks / numlocal;
```

Comment:
Is this the actual bugfix? I'm trying to understand if the socket init changes are necessary, or if they are an orthogonal optimization.

Reply:
That is the actual bug fix.